tests/Feature/ObservationSharingTest.php

```diff
@@ -0,0 +1,165 @@
+
+        $token = ShareToken::where('user_id', $user->id)
+            ->where('observation_id', $observation->id)
+            ->first();
+
+        $response->assertSee($token->value);
+        $response->assertStatus(200);
+    }
+
+    /**
+     * Test invalid token returns fuzzified location.
+     *
+     * @test
+     */
+    public function testInvalidTokenShowsInaccurateLocation()
+    {
+        $user = factory(User::class)->create([
+            'role_id' => Role::where('name', 'User')->first()->id,
+        ]);
+
+        $owner = factory(User::class)->create();
+        $observation = factory(Observation::class)->create([
+            'user_id' => $owner->id,
+        ]);
+
+        $this->actingAs($user);
+
+        $response = $this->get("/web/observation/$observation->id?token=123");
+
+        $json = $response->json();
+
+        $this->assertEquals($json['data']['location']['longitude'], $observation->fuzzy_coords['longitude']);
+        $this->assertEquals($json['data']['location']['latitude'], $observation->fuzzy_coords['latitude']);
+
+        $response->assertStatus(200);
+    }
+
+    /**
+     * Test valid token returns correct location for non-owners.
+     *
+     * @test
+     */
+    public function testValidTokenShowsAccurateLocationToNonOwner()
+    {
+        $owner = factory(User::class)->create();
+
+        $user = factory(User::class)->create([
+            'role_id' => Role::where('name', 'User')->first()->id,
+        ]);
+
+        $observation = factory(Observation::class)->create([
+            'user_id' => $owner->id,
+        ]);
+
+        $token_value = Str::random(60);
+
+        $token = factory(ShareToken::class)->create([
+            'user_id' => $owner->id,
+            'observation_id' => $observation->id,
+            'value' => $token_value,
+        ]);
+
+        $this->actingAs($user);
+
+        $response = $this->get("/web/observation/$observation->id?token=$token->value");
+
+        $json = $response->json();
+
+        $this->assertEquals($json['data']['location']['longitude'], $observation->longitude);
+        $this->assertEquals($json['data']['location']['latitude'], $observation->latitude);
+
+        $response->assertStatus(200);
+    }
+
+    /**
+     * Test expired token returns inaccurate location.
+     *
+     * @test
+     */
+    public function testExpiredTokenShowsInaccurateLocation()
+    {
+        $user = factory(User::class)->create([
+            'role_id' => Role::where('name', 'User')->first()->id,
+        ]);
+
+        $owner = factory(User::class)->create();
+        $observation = factory(Observation::class)->create([
+            'user_id' => $owner->id,
+        ]);
+
+        $token_value = Str::random(60);
+
+        $token = factory(ShareToken::class)->create([
+            'user_id' => $owner->id,
+            'observation_id' => $observation->id,
+            'value' => $token_value,
+            'expired_at' => Carbon::now(),
```

Comment:
Makes sure it's expired:
```suggestion
            'expired_at' => Carbon::now()->subDays(2),
```